### ISSUE TITLE
ApiGateway EndpointConfiguration not required + python 2 should fail (but does not)

### ISSUE THREAD

**Issue by vStone** [cloudtools/troposphere#938]
I'm running into a weird error when trying to update our code base to python 3.5:

* Test script:
```python
#!/usr/bin/python
from troposphere import Template, apigateway
import sys
template = Template(Description="Test Python {}.{}".format(sys.version_info.major, sys.version_info.minor))
lambda_api_domain = template.add_resource(apigateway.DomainName(
    "LambdaApiDomain",
    CertificateArn='foobar',
    DomainName='foobar',
))
print(template.to_json())
```

* 2.7 output:
```
{
    "Description": "Test Python 2.7",
    "Resources": {
        "LambdaApiDomain": {
            "Properties": {
                "CertificateArn": "foobar",
                "DomainName": "foobar"
            },
            "Type": "AWS::ApiGateway::DomainName"
        }
    }
}
```

* 3.5 output:

```
Traceback (most recent call last):
  File "./test3.5.py", line 9, in <module>
    print(template.to_json())
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 623, in to_json
    return json.dumps(self.to_dict(), indent=indent,
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 620, in to_dict
    return encode_to_dict(t)
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 66, in encode_to_dict
    props[name] = encode_to_dict(prop)
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 66, in encode_to_dict
    props[name] = encode_to_dict(prop)
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 57, in encode_to_dict
    return encode_to_dict(obj.to_dict())
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 234, in to_dict
    self._validate_props()
  File "/home/jan/.local/lib64/python3.5/site-packages/troposphere/__init__.py", line 299, in _validate_props
    raise ValueError(msg)
ValueError: Resource EndpointConfiguration required in type AWS::ApiGateway::DomainName (title: LambdaApiDomain)
```

**Comment by ikben** [comment]
It looks like there are two problems here:

- Troposphere considers EndpointConfiguration as required, but it isn't by CloudFormation ([troposphere](https://github.com/cloudtools/troposphere/blob/master/troposphere/apigateway.py#L179) - [CloudFormation](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-domainname.html#cfn-apigateway-domainname-endpointconfiguration)).
- Python3 does enforce this (as it should), but python2 does not.

**Comment by markpeek** [comment]
When I took your sample and ran it on python 2.7 I had the same error/traceback as your 3.5. Please check to see if you have two different versions of troposphere installed.
```
>>> import troposphere
>>> print(troposphere.__version__)
2.1.2
```
Per @ikben it does appear the EndpointConfiguration property for DomainName is incorrect and will be fixed.

**Comment by markpeek** [comment]
Thank you for the issue. Fixed via 7d374672d8bbc84fe514985198a8ce5f2a73800f
